modules/cart/tests/modules/commerce_cart_test/commerce_cart_test.module

```diff
@@ -0,0 +1,19 @@
+<?php
+
+use Drupal\commerce_order\Entity\OrderInterface;
+use \Drupal\Core\Entity\EntityInterface;
+
+/**
+ * Implements hook_entity_update().
+ *
+ * To replicate editor.module and catch save inceptions.
+ */
+function commerce_cart_test_entity_update(EntityInterface $entity) {
+  // Only act on order entities.
+  if (!($entity instanceof OrderInterface)) {
+    return;
+  }
+  if (!$entity->isNew() && empty($entity->original)) {
+    throw new \Exception('Found stale data! $entity->original must be present on updated entity!');
```

Comment:
This doesn't have to mean that the data is stale. Better to say '$entity->original not found', and perhaps make that exception EntityMalformedException?